### PR TITLE
unified-api: adds SentryHub and Static API

### PR DESCRIPTION
**Pull request by fetzig**
* updated travis osx_image to most current (xcode11.2)
 * travis osx_image update resulted in most lanes not working anymore. Many "redefine modules" errors.
 * fixed this by fixing the only non "rediefine module" error in `SentryCrashMemory.c`, which seem to have fixed all travis lanes.
 * fixed/tweaked/checked all travis lanes.
 * adds unified static API `SentrySDK` and `SentryHub` as far as useful and possible without breaking the old API (further work will at least partly break old API, i.e. SentryClient.sharedClient)
 * dropped oclint target/scheme since it doesn't work reliably. Contrary to fastlane oc_lint lane. Which works consistently. 
 * set version number to 5.0.0
 * added some tests for `SentrySDK` and `SentryHub`. Mainly course of sure code coverage.

### CODE REVIEW

File: Sources/Sentry/SentryHub.m [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    static SentryHub *_sharedInstance = nil;
+    static dispatch_once_t oncePredicate;
+    dispatch_once(&oncePredicate, ^{
+        _sharedInstance = [[self alloc] init];
+    });
+     
+    return _sharedInstance;
+}
+
+- (void)startWithOptions:(NSDictionary<NSString *,id> *)options {
```
Comment (HazAT) [inline Sources/Sentry/SentryHub.m]: This should be in `startWithOptions` in `SentrySDK` this function shouldn't exist here.
There can be an `init` that takes a `client` and a `scope`.

`init` doesn't have to be in this PR but the logic should still live in `SentrySDK`

File: Sources/Sentry/include/SentryHub.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+NS_ASSUME_NONNULL_BEGIN
+@interface SentryHub : NSObject
+
++ (instancetype)defaultHub;
+
+- (void)startWithOptions:(NSDictionary<NSString *,id> *)options;
+
+// keeps a stack of client and scope
+
+//- `Hub::new(client, scope)`: Creates a new hub with the given client and scope. The client can be reused between hubs. The scope should be owned by the hub (make a clone if necessary)
```
Comment (HazAT) [inline Sources/Sentry/include/SentryHub.h]: In the final version we should write proper comments, I suppose these are dev helpers.

Comment (fetzig) [reply Sources/Sentry/include/SentryHub.h]: yes. but still, will fix this now.

File: Sources/Sentry/SentryHub.m (hunk 1 above)
Comment (fetzig) [reply Sources/Sentry/SentryHub.m]: yeah. that's right. will move this to `SentrySDK`. Still, I would like to detach `client` and `scope` from each other in a PR dedicated to this task. So my changes will move startWithOptions to `SentrySDK`, but the `SentryHub.init` will take just the client for now.

at least thats my best idea for now.

File: Sources/Sentry/include/SentryHub.h [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+//- `Hub::configure_scope(callback)`: Invokes the callback with a mutable reference to the scope for modifications This can also be a `with` statement in languages that have it (Python).
+
+/**
+ Adds a breadcrumb to the current client.
+ TODO(fetzig): add it to scope instead, once we have it.
+ */
+- (void)addBreadcrumb:(SentryBreadcrumb *)crumb;
+- (void)addBreadcrumbs:(NSArray<SentryBreadcrumb *> *)crumblist;
+- (void)addBreadcrumbWithBlock:(SentryBreadcrumb *(^)(void))block;
+- (void)addBreadcrumbsWithBlock:(NSArray<SentryBreadcrumb *> *(^)(void))block;
```
Comment (HazAT) [inline Sources/Sentry/include/SentryHub.h]: We can remove those functions or at least mark them to be removed.
Generally, to less API the better :)

Comment (fetzig) [reply Sources/Sentry/include/SentryHub.h]: +1
done

**Pull request merged**